### PR TITLE
 keep the pcr right in ts files

### PR DESCRIPTION
**Pull request by yangyijie88**

### CODE REVIEW

File: trunk/src/app/srs_app_hls.cpp [hunk 1]
```diff
@@ -231,9 +232,25 @@ class SrsMpegtsWriter
             // continuity_counter; //4bits
             *p++ = 0x10 | (frame->cc & 0x0f);
             
+            if(first&&frame->key&&is_video)
```
Comment (winlinvip) [inline trunk/src/app/srs_app_hls.cpp]: 这段代码的意图是：每隔几个video就写入pcr？

Comment (yangyijie88) [inline trunk/src/app/srs_app_hls.cpp]: 是啊，比较笨。不过我看到ffmpeg的想法差不多。

**Comment by yangyijie88** [comment]
这是ffmpeg的代码：
if (ts_st->pid == ts_st->service->pcr_pid) {
            if (ts->mux_rate > 1 || is_start) // VBR pcr period is based on frames
                ts_st->service->pcr_packet_count++;
            if (ts_st->service->pcr_packet_count >=
                ts_st->service->pcr_packet_period) {
                ts_st->service->pcr_packet_count = 0;
                write_pcr = 1;
            }
        }

        if (ts->mux_rate > 1 && dts != AV_NOPTS_VALUE &&
            (dts - get_pcr(ts, s->pb)/300) > delay) {
            /* pcr insert gets priority over null packet insert */
            if (write_pcr)
                mpegts_insert_pcr_only(s, st);
            else
                mpegts_insert_null_packet(s);
            continue; /* recalculate write_pcr and possibly retransmit si_info */
        }

**Comment by winlinvip** [comment]
OK, 明白了，我会在2.0考虑合并这段代码。
HLS这块我会花时间好好看看。

**Comment by winlinvip** [comment]
#297

**Pull request closed without merging**